LICENSE.md

```diff
@@ -67,58 +67,59 @@ The User may not limit access to the Mod in any way, except as provided by the G
+Unless specified by a seperate Contributor License Agreement, the Contributor agrees to the following terms.  
+If the contribution is accepted by the Author:
+ - Full copyright of the contribution is assigned to the Author.
+ - The author grants the Contributor a license to use, modify, and distribute the contribution.
 
-7. Right to derive
+7. Right to create and distribute Derivatives
 --------
 
-The User may create Derivative(s) based on the Mod.  
-A Derivative must contain changes which a reasonably informed person would consider significant.  
-
-The User may distribute a Derivative if the following conditions are met:
- - The Derivative does not generate revenue.
- - The Derivative provides credit to the Author.
- - The Derivative complies with the Mod license.
+The User may create and distribute a Derivative of the Mod if all of the following conditions are met:
+ - The Derivative must contain changes which a reasonably informed person would consider significant.  
+ - The Derivative must carry a prominent notice that it is an altered version the Mod.
+ - The Derivative must not generate revenue.
+ - The Derivative must attribute the Author.
+ - The Derivative must be distributed under this same license, or any later version.
+The right to distribute a Derivative that generates revenue must be granted in a separate license which must be explicitly obtained from the Author.
 
-8. Right to distribute addons
+8. Right to create and distribute Addons
 --------
 
-The User may develop non-Derivative Addons.  
-An Addon may include the Mod API without being considered a Derivative.  
-The Addon may be distributed under a different license.  
+The User may create and distribute an Addon to the Mod without restriction.  
+In addition:
+ - The Addon may include the Mod API without being considered a Derivative.  
+ - The Addon may be distributed under a different license.  
 
-9. Right to create Modpack(s)
+9. Right to create and distribute Modpacks
 --------
 
-The User may create Modpack(s) which include the Mod.  
-
-Modpacks containing the Mod may not be used to generate revenue unless specifically granted this right by the Author.  
-
-Modpacks containing the Mod may be made and distributed by the Modpack Creator through any distribution channel they choose.  
+The User may create and distribute a Modpack that contains the Mod through any distribution channel provided the following conditions are met:
+ - The Modpack must not generate revenue.
+ - Any call to download or otherwise aquire the Modpack must be accompanied by a notice stating:
+   - The Modpack contains the Mod.
+   - The Mod was created by the Author.
+   - The URL of the Author's website, if such a website exists.
+The right to distribute a Modpack that generates revenue must be granted in a separate license which must be explicitly obtained from the Author.
```

Comment:
Again, requiring a separate license is something I'm against. Permission is permission...it can be as simple as "yes, you put your modpack behind an AdFly link", or as complex as the Author likes.